apm-agent-plugins/apm-dubbo-plugin/src/main/java/co/elastic/apm/agent/dubbo/advice/AlibabaMonitorFilterAdvice.java

```diff
@@ -57,77 +68,54 @@ public static void init(ElasticApmTracer tracer) {
     public static void onEnterFilterInvoke(@Advice.Argument(1) Invocation invocation,
                                            @Advice.Local("span") Span span,
                                            @Advice.Local("apiClazz") Class<?> apiClazz,
-                                           @Advice.Local("transaction") Transaction transaction,
-                                           @Advice.Local("scope") Scope scope) {
+                                           @Advice.Local("transaction") Transaction transaction) {
         RpcContext context = RpcContext.getContext();
         AlibabaDubboAttachmentHelper helper = helperManager.getForClassLoaderOfClass(Invocation.class);
-        if (helper == null) {
+        if (helper == null || tracer == null) {
             return;
         }
         // for consumer side, just create span, more information will be collected in provider side
         if (context.isConsumerSide()) {
-            if (tracer == null || tracer.getActive() == null) {
+            if (tracer.getActive() == null) {
                 return;
             }
-            span = DubboTraceHelper.createConsumerSpan(invocation.getInvoker().getInterface(),
+            span = DubboTraceHelper.createConsumerSpan(tracer, invocation.getInvoker().getInterface(),
                 invocation.getMethodName(), context.getRemoteAddress());
             if (span != null) {
                 span.getTraceContext().setOutgoingTraceContextHeaders(invocation, helper);
             }
-
-            return;
+        } else {
+            // for provider side
+            transaction = tracer.startChildTransaction(invocation, helper, Invocation.class.getClassLoader());
+            if (transaction != null) {
+                transaction.activate();
```

Comment:
`transaction.activate()` may cause a concurrent problem, the transaction can be ended properly, and no data will be reported to apm server.